### PR TITLE
Mals 130 validate sleepuntil twi match

### PR DESCRIPTION
**Pull request by cgjeffries**
### Linked Jira Issue:
https://ser401-2022-group39.atlassian.net/browse/MALS-130

### Description of Changes:
Validated the testing program for waking on twi match, and added a readme with instructions, as well as example current draw measurements.

### CODE REVIEW

File: megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+8. Observe that the curiosity nano sends messages when entering various sleep modes, as well as when the system receives a valid address over TWI.
+9. Open a serial console in the COM port that the Arduino uno compatible board is on. Set the Baud rate to **57600**.
+10. Observe that the Arduino uno compatible board sends "Sending message to slave!" every 2 seconds.
+11. Verify using the ammeter that the system enters sleep, and then exits sleep upon the receipt of a valid TWI address, by default every 2 seconds. 
+Example:
+![right_address.png](right_address.png)
+12. Revise the `twiAddress` in the Master sketch to a valid value other than 0x08, such as 0x0A.
+13. Download the master sketch to the arduino uno compatible board.
+14. Observe using the ammeter and the Curiosity nano board's serial output that it no longer wakes from sleep, as teh address is not correct. If the ammeter is sufficiently fast reacting, you may see small power spikes as the system received the invalid address.
```
Comment (jnbrauer) [inline megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md:20]:
```suggestion
14. Observe using the ammeter and the Curiosity nano board's serial output that it no longer wakes from sleep, as the address is not correct. If the ammeter is sufficiently fast reacting, you may see small power spikes as the system received the invalid address. 
```

File: megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# `sleepUntilTWIMatch` Test Case
+This test case validates the ability for the system to wake when the correct address is received over TWI
+
+## Instructions
+1. Connect the I2C connections of the Curiosity nano board and an Arduino uno compatible board, such as an arduino nano, arduino uno, or atmega328p Xplained mini. No external pull-up resistors are required. Ensure that the Ground pins of each board are connected,
```
Comment (jnbrauer) [inline megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md:5]:
```suggestion
1. Connect the I2C connections of the Curiosity Nano board and an Arduino Uno compatible board, such as an Arduino Nano, Arduino Uno, or ATmega328p XPlained Mini. No external pull-up resistors are required. Ensure that the ground pins of each board are connected,
```

File: megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+2. Connect the Curiosity nano board to your computer
```
Comment (jnbrauer) [inline megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md:6]:
```suggestion
2. Connect the Curiosity Nano board to your computer
```

File: megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+3. Connect the curiosity nano power supply in series with an ammeter such as a multimeter or AVR power debugger (recommended)
```
Comment (jnbrauer) [inline megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md:7]:
```suggestion
3. Connect the Curiosity Nano power supply in series with an ammeter such as a multimeter or AVR power debugger (recommended)
```

File: megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+4. Download the "slave" sketch to the Curiosity nano board.
```
Comment (jnbrauer) [inline megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md:8]:
```suggestion
4. Download the "slave" sketch to the Curiosity Nano board.
```

File: megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+5. Connect the Arduino uno compatible board to your computer
```
Comment (jnbrauer) [inline megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md:9]:
```suggestion
5. Connect the Arduino Uno compatible board to your computer
```

File: megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+6. Download the "master" sketch to the uno compatible board.
```
Comment (jnbrauer) [inline megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md:10]:
```suggestion
6. Download the "master" sketch to the Uno compatible board.
```

File: megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+7. Open a serial console in the COM port that the Curiosity nano is on, either using the built-in serial console in the Arduino IDE or using a separate program like PuTTY. Set the Baud rate to **115200**.
```
Comment (jnbrauer) [inline megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md:11]:
```suggestion
7. Open a serial console in the COM port that the Curiosity Nano is on, either using the built-in serial console in the Arduino IDE or using a separate program like PuTTY. Set the Baud rate to **115200**.
```

File: megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+8. Observe that the curiosity nano sends messages when entering various sleep modes, as well as when the system receives a valid address over TWI.
```
Comment (jnbrauer) [inline megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md:12]:
```suggestion
8. Observe that the Curiosity Nano sends messages when entering various sleep modes, as well as when the system receives a valid address over TWI.
```

File: megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+9. Open a serial console in the COM port that the Arduino uno compatible board is on. Set the Baud rate to **57600**.
```
Comment (jnbrauer) [inline megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md:13]:
```suggestion
9. Open a serial console in the COM port that the Arduino Uno compatible board is on. Set the Baud rate to **57600**.
```

File: megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+10. Observe that the Arduino uno compatible board sends "Sending message to slave!" every 2 seconds.
+11. Verify using the ammeter that the system enters sleep, and then exits sleep upon the receipt of a valid TWI address, by default every 2 seconds. 
+Example:
+![right_address.png](right_address.png)
+12. Revise the `twiAddress` in the Master sketch to a valid value other than 0x08, such as 0x0A.
```
Comment (jnbrauer) [inline megaavr/extras/testCases/sleepUntilTWIMatch_testCase/README.md:18]:
```suggestion
12. Revise the `twiAddress` in the master sketch to a valid value other than 0x08, such as 0x0A.
```

**Review by jnbrauer — CHANGES_REQUESTED** [review]

**Comment by cgjeffries** [comment]
@jnbrauer committed your suggestions, thank you.

**Review by jnbrauer — APPROVED** [review]
Looks good!

**Pull request merged**